pyodata/v2/model.py

```diff
@@ -2064,21 +2076,56 @@ def __init__(self):
         super(Edmx, self).__init__()
 
     @staticmethod
-    def parse(metadata_xml):
+    def parse(metadata_xml, namespaces=None):
         """ Build model from the XML metadata"""
         if isinstance(metadata_xml, str):
             mdf = io.StringIO(metadata_xml)
         elif isinstance(metadata_xml, bytes):
             mdf = io.BytesIO(metadata_xml)
         else:
             raise TypeError('Expected bytes or str type on metadata_xml, got : {0}'.format(type(metadata_xml)))
-        # the first child element has name 'Edmx'
-        edmx = etree.parse(mdf)
+
+        NAMESPACES['edmx'] = None
+        NAMESPACES['edm'] = None
+        del NAMESPACES['edmx']
+        del NAMESPACES['edm']
+
+        if namespaces is not None:
+            NAMESPACES.update(namespaces)
+
+        xml = etree.parse(mdf)
+        edmx = xml.getroot()
+
+        try:
+            dataservices = next((child for child in edmx if etree.QName(child.tag).localname == 'DataServices'))
+        except StopIteration:
+            raise PyODataParserError('Metadata document is missing the element DataServices')
```

Comment:
Is this line covered in our tests?

Reply:
Line 859

Reply:
fce test_missing_data_service (line 813)

Reply:
OMG, sorry.

Reply:
No worries. Is it otherwise ok?

Reply:
OK